stakingModel/contracts/UBIScheme.sol

```diff
@@ -67,52 +62,42 @@ contract UBIScheme is AbstractUBI {
         maxInactiveDays = _maxInactiveDays;
     }
 
-    function start() public onlyRegistered {
-        super.start();
-        setDay();
-        _withdrawFromDao();
-        DAOToken token = avatar.nativeToken();
-        emit UBIStarted(token.balanceOf(address(this)), now);
-    }
-
     /* @dev On a daily basis UBIScheme withdraw tokens from GoodDao.
      * Emits event with caller address and last day balance and the
      * updated balance.
      */
     function _withdrawFromDao() internal
     {
         DAOToken token = avatar.nativeToken();
-        uint256 lastDayBalance = (dailyFunds[currentDay.sub(1)].openAmount).sub(claimDay[currentDay.sub(1)].claimAmount);
-        require(lastDayBalance == token.balanceOf(address(this)), "Balance does not match with stats");
+        uint256 prevBalance = token.balanceOf(address(this));
         uint256 toWithdraw = token.balanceOf(address(avatar));
         controller.genericCall(
             address(token),
             abi.encodeWithSignature("transfer(address,uint256)", address(this), toWithdraw),
             avatar,
             0);
-        uint256 balance = lastDayBalance.add(toWithdraw);
-        require(balance == token.balanceOf(address(this)), "DAO transfer has failed");
+        uint256 newBalance = prevBalance.add(toWithdraw);
+        require(newBalance == token.balanceOf(address(this)), "DAO transfer has failed");
```

Comment:
I don't think it will fail. It's an atomic test:
`prevBalance` - token balance before the transfer from the avatar (not from stats)
`toWithdraw` - current avatar token balance (not stats)
`newBalance` - `prevBalance` + `toWithdraw` => should be equal to the balance after the transfer from the avatar

Reply:
why do we need it?

Reply:
making sure that the funds have transferred.